metagraph/src/graph/representation/canonical_dbg.cpp

```diff
@@ -48,13 +65,13 @@ uint64_t CanonicalDBG::num_nodes() const {
 void CanonicalDBG
 ::add_sequence(std::string_view sequence,
                const std::function<void(node_index)> &on_insertion) {
-    if (!graph_ptr_)
-        throw std::runtime_error("add_sequence only supported for non-const graphs.");
+    assert(graph_ptr_ && "add_sequence only supported for non-const graphs.");
```

Comment:
Are you sure this is checked somewhere in parents and there is no way to get this wrong with certain commands in CLI?

`assert` must only be called only if there is absolutely no way to get that error once it's compiled.

Reply:
I know. As far as I know, the `add_sequence` is only called in build, augment, and when constructing a query graph. CanonicalDBG is never used for these.

Reply:
if you're sure, you can keep the assert.